projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php

```diff
@@ -614,6 +614,7 @@ public function dashboard_switcher_scripts() {
 	 */
 	public function set_preferred_view( $screen, $view ) {
 		$preferred_views            = $this->get_preferred_views();
+		$screen                     = remove_query_arg( array( 'post_type', 'page', 'taxonomy' ), $screen );
```

Comment:
Because the scope of the problem seems limited to `?post_type=post`, I think we should only remove `?post_type=post` from the URL when present (see extended review note)

Reply:
Thank you for the detail test. I know there are a bunch of other screens involve but we seem to only be adding query params for `post_type, taxonomy and page`

Reply:
The issue also happens with pages

Reply:
> Because the scope of the problem seems limited to `?post_type=post`, I think we should only remove `?post_type=post` from the URL when present (see extended review note)

👍

Reply:
@vykes-mac Sorry, I should've been more clear 😦 

With 1852efc160188bfe01062694ebc51c4af88ecf95, the `post_type=page` is _incorrectly_ removed from `edit.php?post_type=page`, which means the bug is introduced in the "All Pages" context:

1. Navigate to "All Pages".
2. Switch to the "Classic view"

<img width="1573" alt="image" src="https://user-images.githubusercontent.com/36432/178845309-ba367d64-f4fd-4b30-bdc6-b26d10820284.png">

3. Observe the "Classic view" load as expected:

<img width="1560" alt="image" src="https://user-images.githubusercontent.com/36432/178845549-1ae44c52-48bb-43ef-9b59-3c903a8f0e49.png">

4. Navigate to "Media".
5. After waiting a moment for the nav menu to fully load, hover over "Pages" to observe "Default view" to unexpectedly appear as the URL:

<img width="1562" alt="image" src="https://user-images.githubusercontent.com/36432/178845830-54317a05-4f61-4476-b803-98f93d77fc08.png">

The expected URL is `/wp-admin/edit.php?post_type=page`.

What's happening under the hood is that the previewed view is saved to the `edit.php` key, which corresponds to Posts:

```
array (
  'edit.php' => 'classic',
)
```

For Pages, the preferred view needs to be stored to `edit.php?post_type=page`.

As such, I _think_ I'd recommend a simple:

```
$screen = str_replace( '?post_type=post', '', $screen );
```

`post_type=post` is the special edge case we need to accommodate.

Reply:
Ah I see, I was under the impression that setting pages to classic view would also maintain classic view for posts. It feels a little bit weird that one section of the app is classic view and another section is default view, I think that experience can be jarring but that's current implementation.